packages/react/src/headless.tsx

```diff
@@ -0,0 +1,168 @@
+import type { Options } from '@markprompt/core';
+import * as Dialog from '@radix-ui/react-dialog';
+import React, { ReactNode, createContext, forwardRef, useContext } from 'react';
+import { ReactMarkdown } from 'react-markdown/lib/react-markdown.js';
+
+import { useMarkprompt } from './useMarkprompt.js';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+export type RootProps = React.ComponentProps<typeof Dialog.Root> & {
+  children: React.ReactNode;
+  projectKey: string;
+} & Options;
+
+type State = {
+  answer: string | undefined;
+  loading: boolean;
+  prompt: string;
+  references: string[];
+};
+
+type Actions = {
+  handlePromptChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+};
+
+const MarkpromptContext = createContext<State & Actions>({
+  answer: undefined,
+  loading: false,
+  prompt: '',
+  references: [],
+  handlePromptChange: noop,
+  handleSubmit: noop,
+});
+
+function Root(props: RootProps) {
+  const contextValue = useMarkprompt({
+    projectKey: props.projectKey,
+    completionsUrl: props.completionsUrl,
+    placeholder: props.placeholder,
+    iDontKnowMessage: props.iDontKnowMessage,
+    referencesHeading: props.referencesHeading,
+    loadingHeading: props.loadingHeading,
+    includeBranding: props.includeBranding,
+    model: props.model,
+    promptTemplate: props.promptTemplate,
+    temperature: props.temperature,
+    topP: props.topP,
+    frequencyPenalty: props.frequencyPenalty,
+    presencePenalty: props.presencePenalty,
+    maxTokens: props.maxTokens,
+  });
+
+  return (
+    <MarkpromptContext.Provider value={contextValue}>
+      <Dialog.Root modal>{props.children}</Dialog.Root>
+    </MarkpromptContext.Provider>
+  );
+}
+
+const Trigger = Dialog.Trigger;
+Trigger.displayName = 'Markprompt.Trigger';
+
+const Portal = Dialog.Portal;
+Portal.displayName = 'Markprompt.Portal';
+
+const Overlay = Dialog.Overlay;
+Overlay.displayName = 'Markprompt.Overlay';
+
+const Content = Dialog.Content;
+Content.displayName = 'Markprompt.Content';
+
+const Close = Dialog.Close;
+Close.displayName = 'Markprompt.Close';
+
+const Title = Dialog.Title;
+Title.displayName = 'Markprompt.Title';
+
+const Description = Dialog.Description;
+Description.displayName = 'Markprompt.Description';
```

Comment:
Wouldn't it confuse users if they are importing all of their components from `@markprompt/react` and they would see differently namespaced components show up in their React DevTools?

eg.

```tsx
<Dialog.Portal>
  <Dialog.Content>
    ...
    <Markprompt.Form>
      <Markprompt.Prompt />
    </Markprompt.Form>
    ...
  </Dialog.Content>
</Dialog.Portal>
```

Wouldn't it be better to present a unified namespace? The idea is not to obfuscate our use of `@radix-ui/react-dialog`, users should still be able to find that out (and I think we should mention it in our docs because the documentation of that package is excellent).

Reply:
I think it’s even more confusing if the display name changes as a side effect of the `@markprompt/react` import.

```js
import '@markprompt/react'
import { Content } from '@radix-ui/react-dialog'

console.log(Content.displayName)
// 'Markprompt.Content'
```

Maybe this makes more sense here then:

```js
import * as Dialog from '@radix-ui/react-dialog'

export function Content(props) {
  return <Dialog.Content {...props} />
}

Content.displayName = 'Markprompt.Content'
```

What I understood after last week’s call, was we wanted something like this:

```js
import * as Dialog from '@radix-ui/react-dialog'
import classNames from 'classnames'

export function Content(props) {
  return <Dialog.Content {...props} className={classNames('markprompt-dialog-content', props.className)} />
}
```

Although after yesterday’s call, I unserstand this isn’t what we’re going for.

Reply:
Yes, fair enough, your first example is not the intention here and would be confusing. The second example makes more sense but would have us wrap everything with a new function, creating additional overhead that isn't necessary. Still, it's probably the best option :)

Reply:
I think Remco's approach makes sense, and the overhead here is not really an issue.